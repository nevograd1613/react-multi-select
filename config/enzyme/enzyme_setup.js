// TODO: Remove this `raf` polyfill once the below issue is sorted
// https://github.com/facebookincubator/create-react-app/issues/3199#issuecomment-332842582
import raf from "./tempPolyfills";

import Enzyme from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";

Enzyme.configure({ adapter: new Adapter() });
