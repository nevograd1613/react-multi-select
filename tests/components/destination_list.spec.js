import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-shallow-renderer";

import WithDestinationList, {
  DestinationList,
} from "../../src/components/destination_list";
import List from "../../src/components/list/items_list";

const CustomComponent = jest
  .fn(() => <div>Custom Component</div>)
  .mockName("mockedComponent");

const custom_messages = {
  searchPlaceholder: "Find...",
  noItemsMessage: "No entries available...",
  noneSelectedMessage: "Nothing",
  selectedMessage: "Checked",
  selectAllMessage: "Check all",
  clearAllMessage: "Uncheck all",
};

const unselectItems = jest.fn().mockName("unselectItems");
const clearAll = jest.fn().mockName("clearAll");
const ID = 1;

describe("DestinationList", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList />);
    expect(tree).toMatchSnapshot();
  });

  test("custom selectionStatusRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList selectionStatusRenderer={CustomComponent} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom selectedItemRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList selectedItemRenderer={CustomComponent} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom noItemsRenderer", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList noItemsRenderer={CustomComponent} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("custom messages", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList messages={custom_messages} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed unselectItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList unselectItems={unselectItems} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed clearAll", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList clearAll={clearAll} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedIds", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList selectedIds={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<DestinationList selectedItems={[1, 2]} />);
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems with grouping", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList
        selectedItems={[
          { id: 1, label: "item1", group: "group1" },
          { id: 2, label: "item2", group: "group2" },
          { id: 3, label: "item3", group: "group1" },
        ]}
        withGrouping
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed selectedItems with grouping and empty items", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <DestinationList selectedItems={[]} withGrouping />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("passed unselectItems", () => {
    const onClick = jest.fn();
    const component = shallow(<DestinationList unselectItems={onClick} />);
    const list = component.find(List).at(0);
    list.simulate("click", {}, ID);
    expect(onClick).toHaveBeenCalledWith([ID]);
  });
  test("Case include search with destination list", () => {
    const renderer = new ShallowRenderer();
    const custom_messages = { messages: { searchPlaceholder: "Search" } };
    const tree = renderer.render(
      <WithDestinationList
        showSearch={true}
        messages={custom_messages}
        searchIcon="custom_path_icon"
        searchValue="1"
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("Case disabled search with destination list", () => {
    const renderer = new ShallowRenderer();
    const custom_messages = { messages: { searchPlaceholder: "Search" } };
    const tree = renderer.render(
      <WithDestinationList
        showSearch={false}
        messages={custom_messages}
        searchIcon="custom_path_icon"
        searchValue="1"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
