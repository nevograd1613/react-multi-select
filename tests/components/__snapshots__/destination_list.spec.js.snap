// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationList Case disabled search with destination list 1`] = `
<DestinationList
  filteredItems={[]}
  height={400}
  itemHeight={40}
  listRenderer={[Function]}
  messages={
    {
      "messages": {
        "searchPlaceholder": "Search",
      },
    }
  }
  noItemsRenderer={[Function]}
  selectedIds={[]}
  selectedItemRenderer={[Function]}
  selectionStatusRenderer={[Function]}
  withGrouping={false}
/>
`;

exports[`DestinationList Case include search with destination list 1`] = `
<DestinationList
  filteredItems={[]}
  height={400}
  itemHeight={40}
  listRenderer={[Function]}
  messages={
    {
      "messages": {
        "searchPlaceholder": "Search",
      },
    }
  }
  noItemsRenderer={[Function]}
  selectedIds={[]}
  selectedItemRenderer={[Function]}
  selectionStatusRenderer={[Function]}
  withGrouping={false}
>
  <Search
    onChange={undefined}
    searchIcon="custom_path_icon"
    searchPlaceholder="Search..."
    value="1"
  />
</DestinationList>
`;

exports[`DestinationList custom messages 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Uncheck all"
    noneSelectedMessage="Nothing"
    selected={[]}
    selectedMessage="Checked"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage="No entries available..."
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList custom noItemsRenderer 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[MockFunction mockedComponent]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList custom selectedItemRenderer 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[MockFunction mockedComponent]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList custom selectionStatusRenderer 1`] = `
<Column>
  <mockConstructor
    clearAll={undefined}
    clearAllMessage={undefined}
    noneSelectedMessage={undefined}
    selected={[]}
    selectedMessage={undefined}
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList default snapshot 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList passed clearAll 1`] = `
<Column>
  <SelectionStatus
    clearAll={[MockFunction clearAll]}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList passed selectedIds 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={
      [
        1,
        2,
      ]
    }
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList passed selectedItems 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList passed selectedItems with grouping 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList passed selectedItems with grouping and empty items 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;

exports[`DestinationList passed unselectItems 1`] = `
<Column>
  <SelectionStatus
    clearAll={undefined}
    clearAllMessage="Clear All"
    noneSelectedMessage="None Selected"
    selected={[]}
    selectedMessage="selected"
  />
  <ItemsList
    disabled={false}
    height={355}
    isLocked={undefined}
    itemHeight={40}
    items={[]}
    noItemsMessage={undefined}
    noItemsRenderer={[Function]}
    offset={0}
    onClick={[Function]}
    renderer={[Function]}
    selectedIds={[]}
  />
</Column>
`;
